recipe/deploy/prepare.php

```diff
@@ -36,4 +36,26 @@
 
     // Create shared dir.
     run("cd {{deploy_path}} && if [ ! -d shared ]; then mkdir shared; fi");
+
+    // Check and save the remote HEAD/revision and compare it with the existing saved one (if any)
+    // This avoid unnecessary releases when the last commit id matches the existing one (HEAD)
+    $repository  = trim(get('repository'));
+    $revision    = input()->getOption('revision') ?? null;
+    $remoteHead  = $revision ? $revision :
+        trim(run(sprintf('%s ls-remote %s HEAD | tr -d "HEAD"',
+            get('bin/git'), $repository)));
+    if (true === get('check_remote_head') && null == input()->getOption('tag')) {
+        $headPath = trim(get('deploy_path').'/.dep/HEAD');
+        $isRemoteHeadExists = (bool) run('if [ -f '.$headPath.' ]; then echo 1; else echo 0; fi');
```

Comment:
You can use `test()` fn here. Also better to use double quotes `"[ -f $headPath.]"` as it more readable.